### ISSUE TITLE
Panic: mark_link called on Inactive Segment

### ISSUE THREAD

**Issue by spacejam** [spacejam/sled#356]
Using tons of threads + lock_free_delays is turning out to be a good way to mine concurrency bugs.

A segment is being transitioned to Inactive before all threads that may have open reservations for it have finished their operations and reporting of links / replaces. A solution may be to just add an EBR dropper that gets an Arc<Mutex<SegmentAccountant>> and does segment deactivation in a deferred fashion.

This should reproduce the issue in a few seconds:
```
λ cd benchmarks/stress2
λ cargo run --features=lock_free_delays --release -- --burn-in --threads=80 2>&1 | head -50
    Finished release [optimized + debuginfo] target(s) in 0.06s
     Running `target/release/stress2 --burn-in --threads=80`
waiting on signal
did 50828 ops
thread '<unnamed>' panicked at 'assertion failed: `(left == right)`
  left: `Inactive`,
 right: `Active`: expected segment with lsn 1504000000 to be Active', /Users/tylerneely/src/sled/crates/pagecache/src/segment.rs:284:9
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
             at libstd/panicking.rs:475
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:390
   6: std::panicking::try::do_call
             at libstd/panicking.rs:345
   7: pagecache::segment::Segment::lsn
             at /Users/tylerneely/src/sled/crates/pagecache/src/segment.rs:284
   8: pagecache::segment::SegmentAccountant::possibly_clean_or_free_segment
             at /Users/tylerneely/src/sled/crates/pagecache/src/segment.rs:918
   9: pagecache::segment::SegmentAccountant::free_segment
             at /Users/tylerneely/src/sled/crates/pagecache/src/segment.rs:802
  10: pagecache::iobuf::IoBufs::with_sa
             at /Users/tylerneely/src/sled/crates/pagecache/src/pagecache.rs:686
             at /Users/tylerneely/src/sled/crates/pagecache/src/iobuf.rs:222
  11: <pagecache::pagecache::PageCache<PM, P, R>>::cas_page
             at /Users/tylerneely/src/sled/crates/pagecache/src/log.rs:190
             at /Users/tylerneely/src/sled/crates/pagecache/src/pagecache.rs:684
  12: <pagecache::pagecache::PageCache<PM, P, R>>::rewrite_page
             at /Users/tylerneely/src/sled/crates/pagecache/src/pagecache.rs:631
  13: <pagecache::pagecache::PageCache<PM, P, R>>::link
             at /Users/tylerneely/src/sled/crates/pagecache/src/pagecache.rs:496
  14: alloc::alloc::box_free
             at /Users/tylerneely/src/sled/crates/sled/src/tree.rs:378
  15: stress2::run
             at src/main.rs:104
...
```
This may be pretty rare in the wild, but with lock_free_delays + tons of threads we can do a pretty reasonable job with exploring the interleaving space.
